### ISSUE TITLE
KeePass 2.41 Compatibility

### ISSUE THREAD

**Issue by terabyte12** [brush701/keechallenge#38]
It seems KeeChallenge (v1.5) is no longer compatible with the latest KeePass (v 2.41)

**Comment by mrclschstr** [comment]
KeeChallenge v1.5 works fine for me with KeePass v2.41. With KeePass v2.42 and v2.42.1 I have issues unlocking my database, because my Yubikey does not blink and "offer" the key press. The unlocking simply runs into a timeout.

**EDIT:** The KeePass update resetted my slot in the KeeChallenge Settings dialogue. Setting up the correct slot fixed that for me!

**Comment by DevAlphaKilo** [comment]
I am also having issues loading KeeChallenge (v1.5) to load with KeePass (v 2.4.1)

![image](https://user-images.githubusercontent.com/44307521/58192777-91375180-7c8f-11e9-8f76-0a27d223d07f.png)

**Comment by MAP74** [comment]
Same problem here - spent money for a Yubikey which worked well with 2.41. No im stuck to an old version.

@brush701 please give us a life sign if this project is to be continued or if we have to move on

Regards,
Maikl

**Comment by jonasgustavsson** [comment]
I found a workaround. The plugin MultiCipher is based on KeeChallenge, and MultiCipher is compatible with Keepass 2.43.
1. Download the KeePass plugin MultiCipher
1. Open the zip file and extract ONLY the files in the 64bit (or 32bit if you're running that) folder
1. Replace the corresponding files you have in your KeePass installation /plugins folder.
1. Enjoy.

**Comment by jonasgustavsson** [comment]
The trick no longer seems to work with KeePass 2.44. 
I consider KeePass support for YubiKey dead until further.

**Comment by MAP74** [comment]
Its working fine, no workaround needed.
The README itself contains the hint.

No problems at all for the last versions.

From the README

**Common Errors**

Users occasionally report that KeeChallenge does not work with a new version of KeePass. This occurs immediately after an update to KeePass and is easily identified by the warning message: "The following plugin is incompatible with the current KeePass version...". This error is caused by an out of date version of the file KeePass.exe.config, which lives in the KeePass install directory. The problem can be most reliably resolved by doing a complete uninstall/reinstall of KeePass. Alternatively, you can download the portable version of KeePass and copy the config file from there into your KeePass install directory.

**Comment by jonasgustavsson** [comment]
Ah, thanks! That did the trick!
RTFM as one could have said...

**Comment by dschadt** [comment]
using fresh keepass 2.44 portable and keechallenge 1.5 throws the same error on actual W10 64 bit

**Comment by MAP74** [comment]
> 
> 
> using fresh keepass 2.44 portable and keechallenge 1.5 throws the same error on actual W10 64 bit

I'm running the exact same setup and it's working fine.
I also tested your scenario right now with a fresh setup and it works without any issues.

Are you absolutely sure you followed the steps above and you did NOT copy over the file KeePass.exe.config from you production version?

**Comment by dschadt** [comment]
yes, but only on one W10 PC, which is my clients PC to be used in their environment.
Deleted directory in appdata and ../user/../portprog and some regkeys. Copied content of the 2 ZIP fresh from the net to a new dir in portprog and started keepass from there => same error
Installed with same procedure on my own W10 installations => no issue

**Comment by berta54** [comment]
Same error here. I tried it with clean install and also the portable version of KeePass (2.46). Tried the MultiCipher workaround, nothing helped. Did anyone come up with a solution?

**Comment by jonasgustavsson** [comment]
> Same error here. I tried it with clean install and also the portable version of KeePass (2.46). Tried the MultiCipher workaround, nothing helped. Did anyone come up with a solution?

I maintain a portable version of KeePass. When I updated selected files, I did replace most files, but not KeePass.exe.config. When I then tried to start KeePass, it failed with the mentioned error. Replacing also KeePass.exe.config with the version from portable .zip-file resolved the problem.

**Comment by berta54** [comment]
> 
> 
> > Same error here. I tried it with clean install and also the portable version of KeePass (2.46). Tried the MultiCipher workaround, nothing helped. Did anyone come up with a solution?
> 
> I maintain a portable version of KeePass. When I updated selected files, I did replace most files, but not KeePass.exe.config. When I then tried to start KeePass, it failed with the mentioned error. Replacing also KeePass.exe.config with the version from portable .zip-file resolved the problem.

I tried replacing the KeePass.exe.config with the one downloaded from the portable version, but the error remains. Would you maybe share your working KeePass.exe.config file or post the contained text?

**Comment by jonasgustavsson** [comment]
So today I hadn't yet upgraded from KeePass 2.45 to 2.46. For the purpose of this thread, I tried doing the upgrade in the faulty way - not copying the KeePass.exe.config file, but all of the other files. And yes, as expected, I got the compatibility error for KeeChallenge plugin. And as expected, updating the KeePass.exe.config with the 2.46 version solved the issue.

Here's the 2.46 version of KeePass.exe.config. It only differs from the 2.45 one for the values in the bindingRedirect element. (sorry for the formatting)

`<?xml version="1.0" encoding="utf-8" ?>
<configuration>
	<startup useLegacyV2RuntimeActivationPolicy="true">
		<supportedRuntime version="v4.0" />
		<supportedRuntime version="v2.0.50727" />
	</startup>
	<runtime>
		<assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
			<dependentAssembly>
				<assemblyIdentity name="KeePass"
					publicKeyToken="fed2ed7716aecf5c"
					culture="neutral" />
				<bindingRedirect oldVersion="2.0.9.0-2.46.0.0"
					newVersion="2.46.0.18879" />
			</dependentAssembly>
		</assemblyBinding>
		<enforceFIPSPolicy enabled="false" />
		<loadFromRemoteSources enabled="true" />
	</runtime>
	<appSettings>
		<add key="EnableWindowsFormsHighDpiAutoResizing" value="true" />
	</appSettings>
</configuration>`

Also check that you are running the 2.46 version of the .exe file.

**Comment by berta54** [comment]
> 
> 
> So today I hadn't yet upgraded from KeePass 2.45 to 2.46. For the purpose of this thread, I tried doing the upgrade in the faulty way - not copying the KeePass.exe.config file, but all of the other files. And yes, as expected, I got the compatibility error for KeeChallenge plugin. And as expected, updating the KeePass.exe.config with the 2.46 version solved the issue.
> 
> Here's the 2.46 version of KeePass.exe.config. It only differs from the 2.45 one for the values in the bindingRedirect element. (sorry for the formatting)
> 
> `<?xml version="1.0" encoding="utf-8" ?> <configuration> <startup useLegacyV2RuntimeActivationPolicy="true"> <supportedRuntime version="v4.0" /> <supportedRuntime version="v2.0.50727" /> </startup> <runtime> <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1"> <dependentAssembly> <assemblyIdentity name="KeePass" publicKeyToken="fed2ed7716aecf5c" culture="neutral" /> <bindingRedirect oldVersion="2.0.9.0-2.46.0.0" newVersion="2.46.0.18879" /> </dependentAssembly> </assemblyBinding> <enforceFIPSPolicy enabled="false" /> <loadFromRemoteSources enabled="true" /> </runtime> <appSettings> <add key="EnableWindowsFormsHighDpiAutoResizing" value="true" /> </appSettings> </configuration>`
> 
> Also check that you are running the 2.46 version of the .exe file.

Thank you, the KeePass.exe.config seems to be the same as mine. I upgraded to 2.46, did a clean install but the compatibility error keeps showing up. I would like to try it with 2.45 again, but I can't find it anywhere. Do you know where you could get the older legacy versions?

Edit: The same setup with the exact same files does work on my second machine btw, very strange.

**Comment by jonasgustavsson** [comment]
You can find the old releases of KeePass via keepass.info -> Downloads -> Downloads archive, which leads to https://sourceforge.net/projects/keepass/files/KeePass 2.x/

Perhaps you have made the same misstake as I once did with KeePass - Using the Windows start menu, I thought I started KeePass from my portable directory, but in fact it executed another KeePass portable installation I had laying around. That made me utterly confused until I realized my mess.

**Comment by berta54** [comment]
> 
> 
> You can find the old releases of KeePass via keepass.info -> Downloads -> Downloads archive, which leads to https://sourceforge.net/projects/keepass/files/KeePass 2.x/
> 
> Perhaps you have made the same misstake as I once did with KeePass - Using the Windows start menu, I thought I started KeePass from my portable directory, but in fact it executed another KeePass portable installation I had laying around. That made me utterly confused until I realized my mess.

Thank you! I downgraded to 2.45 and now everything works again. The Firefox Kee addon didn't work with 2.46 as well, now both plugins work again. Thanks!

**Comment by sroattle** [comment]
Hi, I don't know if any of you still has this problem. I also ran into it when I tried setting it up today (v 2.52) and installing the "windows login tools" https://www.yubico.com/products/computer-login-tools/ fixed it for me.
I guess this is only a fix for your own devices but not when you use it portable on "external" PCs unless you want to install it every time.
